src/convertToObject.js

```diff
@@ -6,7 +6,55 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.length) {
+    return {};
+  }
+
+  const rawLines = sourceString.split(';');
+
+  // filter lines without letters
+  const noEmptyLines = rawLines
+    .map((line) => {
+      return line.trim();
+    })
+    .filter((item) => item.length > 0);
+
+  const aggregatedLines = noEmptyLines.reduce((prev, line) => {
+    if (line.includes(':')) {
+      prev.push(line);
+    } else {
+      prev[prev.length - 1] += line;
+    }
+
+    return prev;
+  }, []);
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. The `reduce` method here is being used to push items into an array, which is not necessary. You could use a filter or map method to achieve the same result without modifying the accumulator directly.